pages/account/manage/repos.js

```diff
@@ -51,10 +51,13 @@ export default function ManageRepos({ BASE_URL, repos }) {
     additionalMessage: "",
   });
   const [repoList, setRepoList] = useState(repos || []);
-  const [url, setUrl] = useState("");
+  let [url, setUrl] = useState("");
 
   const handleSubmit = async (e) => {
     e.preventDefault();
+    if (url.endsWith("/")) {
+        url = url.substring(0, url.length - 1);
+    }
     const res = await fetch(`${BASE_URL}/api/account/manage/repo`, {
```

Comment:
It's generally not recommended to directly modify state variables. The useState hook provides a setter function (in your case, setUrl) to update the state safely. Maybe we can do something like this ? 

```js
setUrl(url.slice(0, -1));
```

The `setUrl(url.slice(0, -1));` and `url.substring(0, url.length - 1);` is essentially the same, removing the last character.

Reply:
Yeah @tamalCodes, I know about it. But, the `setUrl()` method sets the formatted URL in the text box. So, **the user needs to click on the `Add Repo` button again** which I don't think is a good idea. (**_Please check the below video_**)
Let me know if you have any other options.

[Screencast from 2023-08-13 23-28-20.webm](https://github.com/EddieHubCommunity/LinkFree/assets/105960032/595465f0-8bba-4c9f-abdb-bdf9ddae18f9)

Reply:
I think we could do something like this-
```js
const repoUrl = url.endsWith("/") ? url.slice(0, -1) : url;
```

Reply:
@badrivlog Yes, this can be done. Thank you for the suggestion.
I am going to push this changes as mentioned by @badrivlog.

Reply:
@badrivlog I am facing a runtime error when I try to make your suggested changes as under. Please suggest a fix for this.
```
    const repoUrl = url.endsWith("/") ? url.slice(0, -1) : url;
    const res = await fetch(`${BASE_URL}/api/account/manage/repo`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ repoUrl }),
    });
```
The screenshot of the error is - 
![Screenshot from 2023-08-14 09-38-51](https://github.com/EddieHubCommunity/LinkFree/assets/105960032/c7651d85-ce1d-4aa6-8735-d14bfcd4ab90)